src/figuresClasses.ts

```diff
@@ -1,19 +1,75 @@
 export interface Figure {
-
+  shape: 'triangle'|'circle'|'rectangle';
+  color: 'red'|'green'|'blue';
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
 
-}
+  constructor(
+    public color: 'red'|'green'|'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error("All triangle's sides should be more then zero");
+    }
+
+    if (a >= (b + c) || b >= (a + c) || c >= (b + a)) {
+      throw new Error('Any side must be less than the sum of two other sides');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const sp: number = (this.a + this.b + this.c) / 2;
+    const triangleSquare: number
+     = ((sp * (sp - this.a) * (sp - this.b) * (sp - this.c)) ** 0.5);
 
+    return Math.floor(triangleSquare * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: 'circle' = 'circle';
```

Comment:
Let's create a `type shape` and use it there instead of hardcoding type in each class